core/embed/lib/image_hash_conf.h

```diff
@@ -0,0 +1,50 @@
+#ifndef LIB_IMAGE_HASH_H_
+#define LIB_IMAGE_HASH_H_
+
+#include "model.h"
+#include TREZOR_BOARD
+
+#ifdef IMAGE_HASH_SHA256
+#include "sha2.h"
+#define IMAGE_HASH_DIGEST_LENGTH SHA256_DIGEST_LENGTH
+#ifdef USE_HASH_PROCESSOR
+#include "hash_processor.h"
+#define IMAGE_HASH_CTX hash_sha265_context_t
+#define IMAGE_HASH_INIT hash_processor_sha256_init
+#define IMAGE_HASH_UPDATE hash_processor_sha256_update
+#define IMAGE_HASH_FINAL hash_processor_sha256_final
+#define IMAGE_HASH_CALC hash_processor_sha256_calc
+#else
+#define IMAGE_HASH_CTX SHA256_CTX
+#define IMAGE_HASH_INIT sha256_Init
+#define IMAGE_HASH_UPDATE sha256_Update
+#define IMAGE_HASH_FINAL sha256_Final
+#define IMAGE_HASH_CALC sha256_Raw
+#endif
+
+#elif defined IMAGE_HASH_BLAKE2S
+#include "blake2s.h"
+#define IMAGE_HASH_DIGEST_LENGTH BLAKE2S_DIGEST_LENGTH
+
+static inline void _blake2s_Calc(const uint8_t *data, size_t len,
+                                 uint8_t *hash) {
+  blake2s(data, len, hash, BLAKE2S_DIGEST_LENGTH);
+}
+static inline void _blake2s_Init(BLAKE2S_CTX *ctx) {
+  blake2s_Init(ctx, BLAKE2S_DIGEST_LENGTH);
+}
+static inline void _blake2s_Final(BLAKE2S_CTX *ctx, void *out) {
+  blake2s_Final(ctx, out, BLAKE2S_DIGEST_LENGTH);
+}
+
+#define IMAGE_HASH_CTX BLAKE2S_CTX
+#define IMAGE_HASH_INIT _blake2s_Init
+#define IMAGE_HASH_UPDATE blake2s_Update
+#define IMAGE_HASH_FINAL _blake2s_Final
+#define IMAGE_HASH_CALC _blake2s_Calc
```

Comment:
this is kind of weird style, i would expect something like
```suggestion
#define IMAGE_HASH_CALC(data, len, hash) blake2s(data, len, hash, BLAKE2S_DIGEST_LENGTH)
```
etc.
esp. given that `blake2s_Update` is used raw

is your version better somehow? or does it not matter?

Reply:
my version just sucks https://github.com/trezor/trezor-firmware/pull/3466/commits/3b9962c3baa42201ff7b08c1abcc7eb86af01ac7